src/Elastic.Apm.AspNetCore/DiagnosticListener/AspNetCoreDiagnosticListener.cs

```diff
@@ -41,9 +41,13 @@ public void OnNext(KeyValuePair<string, object> kv)
 				case "Microsoft.AspNetCore.Hosting.HttpRequestIn.Start":
 					if (_httpContextPropertyFetcher.Fetch(kv.Value) is HttpContext httpContextStart)
 					{
-						var transaction = WebRequestTransactionCreator.StartTransactionAsync(httpContextStart, _logger, _agent.TracerInternal,
+						var newTransaction = WebRequestTransactionCreator.StartTransactionAsync(httpContextStart, _logger, _agent.TracerInternal,
```

Comment:
Similar to above comment, converting the result using `as Transaction` might be a little simpler.

Reply:
Similar issue as described above - so I ended up keeping it.